erpnext/manufacturing/report/production_planning_report/production_planning_report.py

```diff
@@ -0,0 +1,328 @@
+				d.warehouse = self.filters.raw_material_warehouse
+				d.required_qty = d.remaining_qty
+				d.actual_qty = 0
+				row.update(d)
+				self.data.append(row)
+
+	def pick_materials_from_warehouses(self, args, order_data, warehouses):
+		required_qty = args.required_qty
+		for index, warehouse in enumerate(warehouses):
+			if not args.remaining_qty: return
+
+			row = self.get_args()
+
+			key = (args.item_code, warehouse)
+			bin_data = self.bin_details.get(key)
+
+			if bin_data:
+				row.update(bin_data)
+
+			args.actual_qty = 0
+			if bin_data and bin_data.get("actual_qty") > 0:
+				args.actual_qty = (bin_data.get("actual_qty")
+					if (args.required_qty > bin_data.get("actual_qty")) else args.required_qty)
+
+				args.remaining_qty -= args.actual_qty 
+				bin_data["actual_qty"] -= args.actual_qty
+
+			if ((self.mrp_warehouses and (args.actual_qty or index == len(warehouses) - 1)) 
+				or not self.mrp_warehouses):
+				if not self.index:
+					row.update(order_data)
+					self.index += 1
+
+				args.warehouse = warehouse
+				row.update(args)
+				if self.purchase_details.get(key):
+					row.update(self.purchase_details.get(key))
+
+				self.data.append(row)
+
+	def get_args(self):
+		return frappe._dict({
+			"work_order": "",
+			"sales_order": "",
+			"production_item": "",
+			"production_item_name": "",
+			"qty_to_manufacture": "",
+			"produced_qty": ""
+		})
+
+	def get_columns(self):
+		based_on = self.filters.based_on
+
+		self.columns = [{
+			"label": _("Id"),
+			"options": based_on,
+			"fieldname": "name",
+			"fieldtype": "Link",
+			"width": 80
+		}, {
+			"label": _("Item Code"),
+			"fieldname": "production_item",
+			"fieldtype": "Link",
+			"options": "Item",
+			"width": 90
+		}, {
+			"label": _("Item Name"),
+			"fieldname": "production_item_name",
+			"fieldtype": "Data",
+			"width": 100
+		}, {
+			"label": _("Warehouse"),
+			"options": "Warehouse",
+			"fieldname": "for_warehouse",
+			"fieldtype": "Link",
+			"width": 100
+		}, {
+			"label": _("Order Qty"),
+			"fieldname": "qty_to_manufacture",
+			"fieldtype": "Float",
+			"width": 80
+		}, {
+			"label": _("Available"),
+			"fieldname": "available_qty",
+			"fieldtype": "Float",
+			"width": 80
+		}, {
+			"label": _("BOM"),
+			"options": "BOM",
+			"fieldname": "bom_no",
+			"fieldtype": "Link",
+			"width": 80
+		},{
+			"label": _("Raw Material"),
+			"fieldname": "item_code",
+			"fieldtype": "Link",
+			"options": "Item",
+			"width": 100
+		}, {
+			"label": _("Item Name"),
+			"fieldname": "item_name",
+			"fieldtype": "Data",
+			"width": 100
+		}, {
+			"label": _("Warehouse"),
+			"options": "Warehouse",
+			"fieldname": "warehouse",
+			"fieldtype": "Link",
+			"width": 100
+		}, {
+			"label": _("Required"),
+			"fieldname": "required_qty",
+			"fieldtype": "Float",
+			"width": 80
+		}, {
+			"label": _("Picked Qty"),
+			"fieldname": "actual_qty",
+			"fieldtype": "Float",
+			"width": 90
+		}, {
+			"label": _("Expected Arrival Data"),
```

Comment:
Expected Arrival Date?

Reply:
This is the required by date() mentioned in the purchase order item, this date helps user to know by when the raw materials will be expected in their warehouse

Reply:
```suggestion
			"label": _("Expected Arrival Date"),
```

Reply:
Fixed